components/automate-ui/src/app/pages/notification-details/notification-details.component.ts

```diff
@@ -0,0 +1,174 @@
+  public hookStatus = UrlTestState.Inactive;
+  private isDestroyed = new Subject<boolean>();
+  notificationDetailsLoading = true;
+
+  constructor(
+    private fb: FormBuilder,
+    private store: Store<NgrxStateAtom>,
+    private layoutFacade: LayoutFacadeService
+  ) {
+
+    this.updateForm = this.fb.group({
+      // Must stay in sync with error checks in data-feed-details.component.html
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      // Note that URL here may be FQDN -or- IP!
+      targetUrl: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]]
+    });
+
+    this.store.select(routeParams).pipe(
+      pluck('id'),
+      filter(identity),
+      takeUntil(this.isDestroyed))
+      .subscribe((id: string) => {
+        this.store.dispatch(new GetNotification({ id }));
+      });
+
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(notificationFromRoute)
+    ]).pipe(
+      filter(([status, notification]) =>
+      status === EntityStatus.loadingSuccess && !isNil(notification)),
+      takeUntil(this.isDestroyed))
+      .subscribe(([_, notification]) => {
+        this.notification = notification;
+        setTimeout(() => (
+          this.updateForm.controls.name.setValue(this.notification.name),
+          this.updateForm.controls.targetUrl.setValue(this.notification.targetUrl),
+          this.notificationDetailsLoading = false
+        ),
+        200);
+      });
+
+    this.store.pipe(
+      select(updateStatus),
+      takeUntil(this.isDestroyed),
+      filter(state => this.saveInProgress && !pending(state)))
+      .subscribe((state) => {
+        this.saveInProgress = false;
+        this.saveSuccessful = (state === EntityStatus.loadingSuccess);
+        if (this.saveSuccessful) {
+          this.updateForm.markAsPristine();
+        }
+      });
+  }
+
+  public ngOnInit(): void {
+    this.layoutFacade.showSidebar(Sidebar.Settings);
+  }
+
+  public saveNotification(): void {
+    this.saveSuccessful = false;
+    this.saveInProgress = true;
+    this.notification.name = this.updateForm.controls['name'].value.trim();
+    this.notification.targetUrl = this.updateForm.controls['targetUrl'].value.trim();
+    this.store.dispatch(new UpdateNotification({
+      notification: this.notification, username: '', password: ''
+    }));
+  }
+
+  public sendTestForTargetUrl(): void {
+    this.testInProgress = true;
+    const name: string = this.updateForm.controls['name'].value.trim();
+    const targetUrl: string = this.updateForm.controls['targetUrl'].value.trim();
+    const targetSecretID: string = this.notification.targetSecretId;
+    this.store.dispatch(new TestNotification({ name, targetUrl, targetSecretID}));
+    this.testInProgress = false;
+  }
+
+  public getAlertTypeKeys() {
+    return this.notification.getAlertTypeKeys();
+  }
+
+  public getTagetTypeKeys() {
+    return this.notification.getTargetTypeKeys();
+  }
```

Comment:
As mentioned earlier, these should be replaced with calculated properties.
Also, there's a typo--should be `getTargetTypeKeys`